app/models/data_resources/event_record.rb

```diff
@@ -29,6 +29,9 @@ class EventRecord < ApplicationRecord
   has_many :dates, as: :dateable, class_name: "FixedDate", dependent: :destroy
   has_one :external_reference, as: :external, dependent: :destroy
 
+  scope :in_date_range, lambda { |start_date, end_date|
+    joins(:dates).where("fixed_dates.date_start >= ? AND fixed_dates.date_start <= ?", start_date, end_date)
```

Comment:
was sind die `fixed_dates`? ist damit gesichert, dass events mit einer längeren laufzeit gefunden werden?

Reply:
abgleich zweimal mit `date_start`? müsste da nicht `date_end` oder so auftauchen?
und dann so logiken wie wen kein ende vorhanden ist etc., wie es in `der list_date` ist? oder bin ich ganz woanders mit meinen gedanken zur date range?

Reply:
mich interessiert ja nur wann die Veranstaltung anfängt und das soll sie in einem Zeitraum von a bis b (von start_date bis end_date)

Reply:
ob das event dann 1 Tag oder 1 Monat geht ist doch egal, oder nicht?

Reply:
geht der fall dann:
- heute ist der 03.01.2021
- eine veranstaltung geht von 01.01.2021 bis 05.01.2021

sehe ich die veranstaltung als heute aktuell?

Reply:
`list_dates` ist eine Methode (glaube sogar von dir) die dir diese Information aus dem Model `fixed_dates` herausarbeitet. Alle Zeitstempel sind meiner Meinung nach in `FixedDate` abgebildet. 

Willst du denn eine Veranstaltung, die 3 Tage geht, aber gestern begonnen hat heute als "buchbar" sehen?
Bisher würde der Fall nicht gehen!

ich bin der Meinung, dass das Ende und die Spanne eines Events für die Suche nicht relevant ist sondern nur das `date_start`.

Wenn du aber auch Events haben möchtest die in der Vergangenheit, vor deinem Zeitsuchfenster, begonnen haben, dann muss man das noch erweitern

Reply:
genau das soll sein. "aktuelle events" sind die, die (auch) heute stattfinden. wenn ein event zeitspanne hat, ist es auch "heute" aktuell und muss returned werden. genau das macht list date ja bereits weshalb ich das im ticket darauf bezog.

Reply:
um "heute" geht es ja nicht, sondern um das 3 Tages Event, das Gestern gestartet ist.

Reply:
@digorath kannst du bitte joinen hier, um zum richtigen ergebnis zu kommen. danke.

wir brauchen aus heutiger sicht alle events, die stattfinden, richtig?

Reply:
ist jetzt umgebaut

Reply:
ansage von Philipp war folgendes:
für das event widget sollen alle events angezeigt/relevant sein, welche a) nur an einem tag statt finden und dieser tag "heute" ist, oder b) mehrtägig sind, und bei welchen "heute" in der mehrtägigen Zeitspanne liegt.

Wenn ich euch richtig verstanden habe, dann stimmt das mit eurem Resultat überein ;)